Adding blog post on Bean Validation 2.0 progress

gunnarmorling opened the pull request.

gunnarmorling committed:
Adding blog post on Bean Validation 2.0 progress

gsmet commented on news/2017-01-19-bean-validation-2-0-progress-report.md:
```diff
@@ -0,0 +1,200 @@
+tags: [news]
+---
+
+It has been a few months since we've kicked of the work on Bean Validation 2.0 (JSR 380).
```
kicked of**f**?

gunnarmorling replied:
Done.

gsmet commented on news/2017-01-19-bean-validation-2-0-progress-report.md:
```diff
@@ -0,0 +1,200 @@
+but also many useful additions to the class library.
+
+One prominent example of the latter is the new date and time API ([JSR 310](https://www.jcp.org/en/jsr/detail?id=310)).
+Types such as `Instant`, `LocalDate` or `ZonedDateTime` are supported by the `@Past` and `@Future` constraints now ([BVAL-496](https://hibernate.atlassian.net/browse/BVAL-496)):
```
I think it would be better to have `now` after `are`: are now supported

gunnarmorling replied:
Done.

gsmet commented on news/2017-01-19-bean-validation-2-0-progress-report.md:
```diff
@@ -0,0 +1,200 @@
+    @Future
+    private LocalDate deliveryDate;
+
+`@Past` and `@Future` also have a new attribute `orPresent()` now:
```
I think 'now also have' reads better.

gunnarmorling replied:
Done.

gsmet commented on news/2017-01-19-bean-validation-2-0-progress-report.md:
```diff
@@ -0,0 +1,200 @@
+
+But it's also useful for (re-)running batch jobs with a different logical date than the current one or for testing with a fixed point in time considered as "now":
+
+    Validator validator = Validation.byDefaultProvider()
```
We don't have syntax highlighting with the triple backticks?

gunnarmorling replied:
Didn't work for me. Converted it to *.adoc.

gsmet commented on news/2017-01-19-bean-validation-2-0-progress-report.md:
```diff
@@ -0,0 +1,200 @@
+    IntegerProperty revenue;
+
+This becomes possible by defining means of "automatic value unwrapping" for specific types such as the JavaFX ones.
+Check out the [latest spec draft](http://beanvalidation.org/latest-draft/spec/#appendix-valueextraction-wrappedelements) to learn more about how this is handled.
```
Hmmm, the spec is currently not in line with the API and the RI.

gunnarmorling replied:
That's true. Some update is due.

gsmet commented on news/2017-01-19-bean-validation-2-0-progress-report.md:
```diff
@@ -0,0 +1,200 @@
+
+We are currently working on picking up some loose ends around the proposal for container element validation.
+Once that is done, we feel it is the right time to put out an Alpha1 release of Bean Validation 2.0 and post it for Early Draft Review to the JCP.
+This should get the discussed changes into the hands of more people out there and will let us improve and hone the features added so far.
```
hone?

gunnarmorling replied:
https://www.merriam-webster.com/dictionary/hone

gsmet commented on news/2017-01-19-bean-validation-2-0-progress-report.md:
```diff
@@ -0,0 +1,200 @@
+<img src="/images/bv_logo_scribble.jpg" alt="Bean Validation logo draft" style="width: 350px" />
+</div>
+
+Could there be a better representation of Bean Validation thank Duke validating a bean?
```
s/thank/than/

gunnarmorling replied:
Done.

gsmet reviewed (commented):
Looks good. Added a couple of comments inline.

Wondering if we shouldn't talk about the remaining challenges of the container thing and add a link to the open questions.

gunnarmorling committed:
Converting progress report post to AsciiDoc

gunnarmorling committed:
Fixing typos as per Guillaume's remarks

gunnarmorling commented:
> Looks good. Added a couple of comments inline.

Thanks for the quick review. Pushed an update for converting to *.adoc and addressing your remarks.

> Wondering if we shouldn't talk about the remaining challenges of the container thing and add a link to the open questions.

I've added the link but didn't dive into details here. I think it's too deep for this post, also it's very long already anyways.

gunnarmorling commented:
It's also on staging if you prefer the actual preview: http://staging.beanvalidation.org/news/2017/01/19/bean-validation-2-0-progress-report/

gsmet commented on news/2017-01-19-bean-validation-2-0-progress-report.adoc:
```diff
@@ -0,0 +1,235 @@
+
+== First changes
+
+With the first EG members on board, we didn't loose time and began with the work on the new spec revision.
```
s/loose/lose/

gsmet commented on news/2017-01-19-bean-validation-2-0-progress-report.adoc:
```diff
@@ -0,0 +1,235 @@
+----
+Putting the constraints to the `String` type argument makes it apparent that they should not be applied to the list object itself, but rather to each contained element.
+
+Similarly, it's possible to put constraints to the elements of an array:
```
Maybe "put constraints on" would be better?

gunnarmorling committed:
Further wording improvements

emmanuelbernard commented on news/2017-01-19-bean-validation-2-0-progress-report.adoc:
```diff
@@ -0,0 +1,235 @@
+    .getValidator();
+----
+
+== Validation of container elements
```
This is not a good title for the audience this blog targets. It needs to mention concrete ones: `Collection` and `Optional`
You can mention later that it is generalized.

gunnarmorling replied:
Ok.

emmanuelbernard commented on news/2017-01-19-bean-validation-2-0-progress-report.adoc:
```diff
@@ -0,0 +1,235 @@
+
+Also cascaded validation gets more flexible with that.
+It's now possible to mandate that the keys _and_ values of maps should be validated
+(so far, only values were validated) by using `@Valid` like this:
```
s/so far/until now/

emmanuelbernard commented on news/2017-01-19-bean-validation-2-0-progress-report.adoc:
```diff
@@ -0,0 +1,235 @@
+ListMultimap<@Valid Customer, @Email String> emailsByCustomer;
+----
+
+We consider to detect custom extractors using the http://docs.oracle.com/javase/8/docs/api/index.html?java/util/ServiceLoader.html[service loader mechanism],
```
We are considering (or contemplating)

emmanuelbernard commented on news/2017-01-19-bean-validation-2-0-progress-report.adoc:
```diff
@@ -0,0 +1,235 @@
+Hence its current proposal is added as http://beanvalidation.org/latest-draft/spec/#appendix-value-extraction[an appendix] to the spec draft.
+We are eager to learn about your thoughts and feedback in general, but it's especially important for this issue due to its complexity.
+We've compiled a list of http://beanvalidation.org/latest-draft/spec/#_open_questions[open questions] around this proposal.
+If you have thoughts on any of those, please make sure to let us know, e.g. by commenting below.
```
If one can run and play with it in the RI, mention it.

emmanuelbernard commented on news/2017-01-19-bean-validation-2-0-progress-report.adoc:
```diff
@@ -0,0 +1,235 @@
+== Next steps
+
+We are currently working on picking up some loose ends around the proposal for container element validation.
+Once that is done, we feel it is the right time to put out an Alpha1 release of Bean Validation 2.0 and post it for Early Draft Review to the JCP.
```
Hum really, we are waiting for feedback and another new cycle before pushing it? I'm against that. Pushing it in the JSP is about feedback.

gunnarmorling replied:
Not waiting really, what I had in mind here is to manifest the latest agreements on some of the open questions in the doc. I'll reword.

emmanuelbernard commented on news/2017-01-19-bean-validation-2-0-progress-report.adoc:
```diff
@@ -0,0 +1,235 @@
+
+We haven't decided yet whether to put this into the spec or not.
+So we recommend you give it a try in the reference implementation and let us know about your thoughts.
+The feedback when https://twitter.com/gunnarmorling/status/819631488358563840[sharing the idea] on Twitter was https://twitter.com/dblevins/status/819633752888475648[very encouraging].
```
You know David has been pushing for it in the first place so that's not surprising

gunnarmorling replied:
Sure, I like the number of favs and retweets, though :)

gunnarmorling committed:
Updating post as per review comments; keeping logo for a separate post
